app/src/main/java/com/uncookthebook/app/LoginFragment.java

```diff
@@ -1,59 +1,76 @@
-import com.google.android.material.textfield.TextInputLayout;
+
+import java.util.Objects;
 
 /**
  * Fragment representing the login screen for Shrine.
  */
 public class LoginFragment extends Fragment {
+    private static final int RC_SIGN_IN = 2;
 
     @Override
     public View onCreateView(
             @NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         // Inflate the layout for this fragment
         View view = inflater.inflate(R.layout.login_fragment, container, false);
-        final TextInputLayout passwordTextInput = view.findViewById(R.id.password_text_input);
-        final TextInputEditText passwordEditText = view.findViewById(R.id.password_edit_text);
-        MaterialButton nextButton = view.findViewById(R.id.next_button);
-
-        // Set an error if the password is less than 8 characters.
-        nextButton.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                if (!isPasswordValid(passwordEditText.getText())) {
-                    passwordTextInput.setError(getString(R.string.error_password));
-                } else {
-                    passwordTextInput.setError(null); // Clear the error
-                    ((NavigationHost) getActivity()).navigateTo(new ReportArticleFragment(), false);
-                }
-            }
-        });
-
-        // Clear the error once more than 8 characters are typed.
-        passwordEditText.setOnKeyListener((view1, i, keyEvent) -> {
-            if (isPasswordValid(passwordEditText.getText())) {
-                passwordTextInput.setError(null); //Clear the error
-            }
-            return false;
-        });
+        MaterialButton nextButton = view.findViewById(R.id.login_button);
+
+        nextButton.setOnClickListener(view1 -> signIn());
 
         return view;
     }
 
-    // "isPasswordValid" from "Navigate to the next Fragment" section method goes here
-    private boolean isPasswordValid(@Nullable Editable text) {
-        return text != null && text.length() >= 8;
+    private void signIn() {
+        GoogleActivity mainActivity = ((GoogleActivity) Objects.requireNonNull(getActivity()));
+        Intent signInIntent = mainActivity.getGoogleClient().getSignInIntent();
+        startActivityForResult(signInIntent, RC_SIGN_IN);
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        // Result returned from launching the Intent from GoogleSignInClient.getSignInIntent(...);
+        if (requestCode == RC_SIGN_IN) {
+            // The Task returned from this call is always completed, no need to attach
+            // a listener.
+            Task<GoogleSignInAccount> task = GoogleSignIn.getSignedInAccountFromIntent(data);
+            handleSignInResult(task);
+        }
```

Comment:
What if the code is not sign_in? What happens? It remains on this fragment without doing nothing?

Reply:
check the code in handleSignInResult